app.rb

```diff
@@ -0,0 +1,116 @@
+class App
+  def initialize()
+    @students = []
+    @teachers = []
+    @classrooms = []
+    @books = []
+    @rentals = []
+    @allpeople = []
+  end
+
+  def choice_teacher
+    print 'Age: '
+    age = gets.to_i
+    print 'Name: '
+    name = gets
+    print 'Specialization : '
+    specialization = gets.chomp
+    require_relative 'teacher'
+    teacher = Teacher.new(age, name, specialization: specialization)
+    @teachers << teacher
+    @allpeople << teacher
+    puts "Teacher created of name #{teacher.name}"
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice_choosed = gets.to_i
+    if choice_choosed == 1
+      print 'Age: '
+      age = gets.to_i
+      print 'Name: '
+      name = gets
+      print 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp.upcase
+      parent_permission = parent_permission == 'Y'
+      require_relative 'student'
+      student = Student.new(age, name, parent_permission: parent_permission)
+      @students << student
+      @allpeople << student
+      puts "Student created of name  #{student.name}"
+    elsif choice_choosed == 2
+
+      choice_teacher
+    else
+      puts 'Invalid option, try again'
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets
+    print 'Author: '
+    author = gets
+    require_relative 'book'
+    book = Book.new(title, author)
+    @books << book
+    puts "Book created of title #{book.title}"
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each do |book|
+      puts "Title: \"#{book.title.chomp}\", Author: \"#{book.author.chomp}\""
+    end
+  end
+
+  def list_all_people
+    puts 'List of People:'
+    @allpeople.each do |person|
+      person_type = person.is_a?(Student) ? '[Student]' : '[Teacher]'
+      puts "#{person_type} Name: #{person.name.chomp}, ID: #{person.id} Age: #{person.age}"
+    end
+  end
+
+  def create_rental
+    puts 'Select a person from the following list by number (not ID):'
+
+    @allpeople.each_with_index do |person, index|
+      person_type = person.is_a?(Student) ? '[Student]' : '[Teacher]'
+      puts "#{index}) #{person_type} Name: #{person.name.chomp}, ID: #{person.id}, Age: #{person.age}"
+    end
```

Comment:
- [ ] Please fix this error your app generates when a user tries to create a rental
<img width="1325" alt="Screenshot 2023-05-27 at 22 16 10" src="https://github.com/byiringiroscar/OOP-school-library-with-Ruby/assets/55596970/a193baba-a183-4119-a66d-b5106161b5c8">